src/config/wmodules-aws.c

```diff
@@ -294,12 +294,6 @@ int wm_aws_read(const OS_XML *xml, xml_node **nodes, wmodule *module)
                 mtdebug2(WM_AWS_LOGTAG, "Creating first service structure");
             }
 
-            // Expand service Child Nodes
-
-            if (!(children = OS_GetElementsbyNode(xml, nodes[i]))) {
-                continue;
-            }
-
             // type is an attribute of the service tag
             if (!strcmp(*nodes[i]->attributes, XML_SERVICE_TYPE)) {
```

Comment:
Control when `nodes[i]->attributes` or `nodes[i]->values` is `NULL`. This happens with this configuration:

```
<service>
</service>
```

And causes SegFault when running the module.

Reply:
Fixed! We have the same problem with buckets configuration.